mlflow/projects/backend/abstract_backend.py

```diff
@@ -0,0 +1,34 @@
+from abc import abstractproperty, abstractmethod, ABCMeta
+
+
+class AbstractBackend():
+    """
+    Abstract class for MLflow Project Execution Backend
+    A backend must implement the run method to launch a new job
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def run(self, active_run, uri, entry_point, parameters,
+            backend_config):
+        """
+        Submit an entrypoint. It must returns a SubmittedRun object to track the execution
+
+        :param active_run: Current MLflow run object. Everything will registred in this run.
```

Comment:
Could `active_run` be replaced with `run_id` containing just the ID of the MLflow run? And then to fetch the experiment ID, individual backends can call `run = mlflow.get_run(run_id); experiment_id = run.info.run_id`? Would be nice to keep the interface simpler for backend implementors (so that they don't necessarily need to learn the fields of `active_run`)

Reply:
done